packages/store/src/patterns/rankOrder.js

```diff
@@ -29,38 +27,48 @@ const { ownKeys } = Reflect;
  */
 const sameValueZero = (x, y) => x === y || is(x, y);
 
+const trivialComparator = (left, right) =>
+  // eslint-disable-next-line no-nested-ternary
+  left < right ? -1 : left === right ? 0 : 1;
+
 /**
- * @type {[PassStyle, RankCover][]}
+ * @typedef {Record<PassStyle, { index: number, cover: RankCover }>} PassStyleRanksRecord
  */
-const PassStyleRankAndCover = harden([
-  /* !  */ ['error', ['!', '!~']],
-  /* (  */ ['copyRecord', ['(', '(~']],
-  /* :  */ ['tagged', [':', ':~']],
-  /* ?  */ ['promise', ['?', '?~']],
-  /* [  */ ['copyArray', ['[', '[~']],
-  /* b  */ ['boolean', ['b', 'b~']],
-  /* f  */ ['number', ['f', 'f~']],
-  /* np */ ['bigint', ['n', 'p~']],
-  /* r  */ ['remotable', ['r', 'r~']],
-  /* s  */ ['string', ['s', 't']],
-  /* v  */ ['null', ['v', 'v~']],
-  /* y  */ ['symbol', ['y', 'z']],
-  /* z  */ ['undefined', ['z', '{']],
-  /* | remotable->ordinal mapping prefix: This is not used in covers but it is
-       reserved from the same set of strings. Note that the prefix is > any
-       prefix used by any cover so that ordinal mapping keys are always outside
-       the range of valid collection entry keys. */
-]);
 
-const PassStyleRank = fromEntries(
-  entries(PassStyleRankAndCover).map(([i, v]) => [v[0], Number(i)]),
+const passStyleRanks = /** @type {PassStyleRanksRecord} */ (
+  fromEntries(
+    entries(passStylePrefixes)
+      // Sort entries by ascending prefix.
+      .sort(([_leftStyle, leftPrefixes], [_rightStyle, rightPrefixes]) => {
+        return trivialComparator(leftPrefixes, rightPrefixes);
+      })
+      .map(([passStyle, prefixes], index) => {
+        // Cover all strings that start with any character in `prefixes`.
+        // `prefixes` is already sorted, so that's
+        // all s such that prefixes.at(0) ≤ s < successor(prefixes.at(-1)).
```

Comment:
Since this is one-time initialization code, would it make sense to assert that `prefixes` is sorted?
```suggestion
        // Cover all strings that start with any character in `prefixes`,
        // verifying that it is sorted so that is
        // all s such that prefixes.at(0) ≤ s < successor(prefixes.at(-1)).
        prefixes === [...prefixes].sort().join('') ||
          assert.fail(X`unsorted prefixes for passStyle ${q(passStyle)}: ${q(prefixes)}`);
```

Reply:
Done